### PR TITLE
Prefer void to null

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: packages/flutter_test/test/widget_tester_test.dart [hunk 1]
```diff
@@ -625,7 +625,7 @@ class FakeMatcher extends AsyncMatcher {
 
   @override
   Future<String> matchAsync(dynamic object) {
-    return completer.future.then<String>((void _) {
+    return completer.future.then<String>((_) {
```
Comment (Hixie) [inline packages/flutter_test/test/widget_tester_test.dart]: these should be `void value`

File: packages/flutter/lib/src/widgets/binding.dart [hunk 2]
```diff
@@ -707,11 +707,11 @@ abstract class WidgetsBinding extends BindingBase with SchedulerBinding, Gesture
   }
 
   @override
-  Future<Null> performReassemble() {
+  Future<void> performReassemble() {
     deferFirstFrameReport();
     if (renderViewElement != null)
       buildOwner.reassemble(renderViewElement);
-    return super.performReassemble().then((Null value) {
+    return super.performReassemble().then((void _) {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/binding.dart]: void value

File: packages/flutter/lib/src/widgets/animated_list.dart [hunk 3]
```diff
@@ -332,7 +332,7 @@ class AnimatedListState extends State<AnimatedList> with TickerProviderStateMixi
         ..sort();
     });
 
-    controller.reverse().then<void>((Null value) {
+    controller.reverse().then<void>((_) {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/animated_list.dart]: void value here and above

File: packages/flutter/lib/src/material/scaffold.dart [hunk 4]
```diff
@@ -1192,7 +1192,7 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
       _snackBarController.value = 0.0;
       completer.complete(reason);
     } else {
-      _snackBarController.reverse().then<void>((Null _) {
+      _snackBarController.reverse().then<void>((_) {
```
Comment (Hixie) [inline packages/flutter/lib/src/material/scaffold.dart]: void value

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 5]
```diff
@@ -301,7 +301,7 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
         scrollOffset < bottomOffset ? bottomOffset : topOffset,
         duration: _scrollAnimationDuration,
         curve: Curves.easeInOut,
-      ).then((Null none) {
+      ).then((void none) {
```
Comment (Hixie) [inline packages/flutter/lib/src/material/reorderable_list.dart]: void value

File: packages/flutter/lib/src/material/refresh_indicator.dart [hunk 6]
```diff
@@ -328,7 +328,7 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
     _mode = _RefreshIndicatorMode.snap;
     _positionController
       .animateTo(1.0 / _kDragSizeFactorLimit, duration: _kIndicatorSnapDuration)
-      .then<void>((Null value) {
+      .then<void>((_) {
```
Comment (Hixie) [inline packages/flutter/lib/src/material/refresh_indicator.dart]: void value

File: packages/flutter/lib/src/cupertino/button.dart [hunk 7]
```diff
@@ -188,10 +186,10 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
     if (_animationController.isAnimating)
       return;
     final bool wasHeldDown = _buttonHeldDown;
-    final Future<Null> ticker = _buttonHeldDown
+    final TickerFuture ticker = _buttonHeldDown
         ? _animationController.animateTo(1.0, duration: kFadeOutDuration)
         : _animationController.animateTo(0.0, duration: kFadeInDuration);
-    ticker.then<void>((Null value) {
+    ticker.then<void>((_) {
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/button.dart]: void value

**Comment by Hixie** [comment]
LGTM modulo minor nits
please update changelog accordingly

**Comment by a14n** [comment]
>please update changelog

Where can I find the changelog?

Regarding `(_)` it seems that `always_specify_types` doesn't lint lambdas.

**Comment by goderbauer** [comment]
You can find the changelog in the wiki. Here's the direct link: https://github.com/flutter/flutter/wiki/Changelog

**Comment by dnfield** [comment]
@a14n - thanks for the contributions.  `flutter-build` was red when this was landed - was this meant to fix the build?   There's another PR that was meant to do that that's still going through the test cycles to make the tree green again (we hope).

**Comment by a14n** [comment]
@dnfield sorry I didn't notice it was red :-1: Should I revert this PR ?

**Comment by a14n** [comment]
Actually I only have access to a small number of tasks in https://flutter-dashboard.appspot.com/build.html and I'm not able to see the cause when it's red. (small excuse: Circus was green when I merged this PR)

**Comment by dnfield** [comment]
No problem, I know it's confusing and sometimes the CI reporting isn't even up to date.  I'd leave it for now.  

For future reference, even if Cirrus is green we want `flutter-build` to be green as well - those are all checks done post commit and run more than just what Cirrus does.  That way we're sure not to pile more to bisect through if we have to revert.

**Comment by a14n** [comment]
>I'm not able to see the cause when it's red

For instance my commit makes the task _channels_integration_test_win_ red but I cannot see why.

>For future reference, even if Cirrus is green we want flutter-build to be green as well - those are all checks done post commit and run more than just what Cirrus does. That way we're sure not to pile more to bisect through if we have to revert.

Ack. Thanks for the information and sorry again for this too early merge

**Comment by dnfield** [comment]
Your commit didn't make channels_integration_test, it was a prior commit doing that.  `flutter-build` looks backwards at landed commits.

**Comment by dnfield** [comment]
Ahh I take it back.  It did make it red.  Unfortunately, those logs are only available internally here.

It was a timeout though, I'm going to rerun - don't think it's related to your code.

**Comment by dnfield** [comment]
Yup was just a flake, think we should be good now - thanks again!

**Comment by a14n** [comment]
Thanks to you :)

**Pull request merged**